### DISCUSSION TITLE
Creating an iOS specific build with Xcode's new multiplatform app template doesn't create .ipa

### DISCUSSION THREAD

**Discussion by knlshh** [fastlane/fastlane#20759] [Q&A]
I'm creating a new SwiftUI app that uses Xcode's new multiplatform app template that supports building macOS + iOS apps in a single codebase. Please note that this is **not** a Catalyst app. 

The Supported Destinations look like this in Xcode:
<img width="1019" alt="Screen Shot 2022-10-19 at 10 22 15 PM" src="https://user-images.githubusercontent.com/10396271/196796061-a0a5d3f9-c61a-4f4f-9210-00715301a5bc.png">

When I run Fastlane's `gym` step with the platform as `ios`, I'd expect the `.ipa` file to be created, but this isn't the case (archiving succeeds, but the `.ipa` file isn't generated).

If I remove the `Mac` destination from the above list and re-run `gym`, the `.ipa` file is created. I suspect this is because of the `building_for_ios` (https://github.com/fastlane/fastlane/blob/950bb820237d52e15690408083e328f739ac0ef3/gym/lib/gym/module.rb#L28) conditional.

Copied as-is from the above link:
```
def building_for_ios?
  if Gym.project.mac?
    # Can be building for iOS if mac project and catalyst
    return building_mac_catalyst_for_ios?
  else
    # Can be iOS project and build for mac if catalyst
    return false if building_mac_catalyst_for_mac?

    # Can be iOS project if iOS, tvOS, or watchOS
    return Gym.project.ios? || Gym.project.tvos? || Gym.project.watchos?
  end
end
```

While determining if Fastlane is building for `ios`, it checks if the project supports `mac` (`Gym.project.mac`), and if yes, returns `building_mac_catalyst_for_ios`. This isn't necessarily the case since a multiplatform app can be building for iOS **and** support Mac **and** still not be building Mac Catalyst for iOS. 

Wondering if my understanding is incorrect or if this is a bug in Fastlane?

**Comment by x-creates** [comment]
Have you found any workaround for now before [https://github.com/fastlane/fastlane/issues/20768](https://github.com/fastlane/fastlane/issues/20768) gets merged?

**Reply by mcolinsk** [reply]
workaround can be done by creating a script to set SUPPORTS_MACCATALYST = 'NO' before doing fastlane gym.

https://github.com/fastlane/fastlane/issues/22276#issuecomment-2807719290
